app/src/main/java/com/example/drawingApp/MainActivity.kt

```diff
@@ -28,7 +22,12 @@ class MainActivity : AppCompatActivity() {
         }
 
         findViewById<TextView>(R.id.newLayer).setOnClickListener {
-            showLayerAlertDialog()
+            DialogObject.showLayerAlertDialog(this, model,
+                onDialogSubmission = { model.layerNames.add(LayerViewModel(it)) }
```

Comment:
This here shows a good pattern for this. Eliminate the need to the model as a dependency